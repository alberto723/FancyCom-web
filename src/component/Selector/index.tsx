import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
// import './style.css';

interface ItemProps {
  id: string;
  label: string;
  icon: object;
}

export interface SelectorProps {
  items: ItemProps[];
  onChange: () => void;
}

export const Selector: React.FC<SelectorProps> = ({ items, onChange }) => {
  const styles: any = useStyles();
  const [value, setValue] = useState(0);
  return (
    <div className={styles.screen}>
      <div className={styles.container}>
        {items &&
          items.map((item, index) => (
            <div
              onClick={() => {
                setValue(index);
                onChange();
              }}
              key={item.id}
              className={`${styles.alignItemContainer}`}
              style={{ backgroundColor: value === index ? '#F4F4F4' : 'white' }}>
              <div className={styles.alignItemSection}>
                <div className={styles.alignIconSection}>{item.icon}</div>
                <p className={styles.alignItemLabel}>{item.label}</p>
              </div>
            </div>
          ))}
      </div>
    </div>
  );
};

const useStyles = makeStyles(() => ({
  screen: {
    flex: 1,
    backgroundColor: 'white',
  },
  container: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  alignItemContainer: {
    paddingLeft: '5px',
    paddingRight: '5px',
    paddingTop: '10px',
    paddingBottom: '10px',
  },
  alignItemSection: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  alignIconSection: {
    display: 'flex',
    width: 60,
    height: 60,
    borderWidth: 2,
    borderColor: '#DCDCDC',
    borderStyle: 'solid',
    alignItems: 'center',
    justifyContent: 'center',
    marginLeft: '5px',
    marginRight: '5px',
  },
  alignItemLabel: {
    color: 'black',
    fontSize: 12,
    fontWeight: 'bold',
    marginTop: '5px',
    textAlign: 'center',
  },
}));
