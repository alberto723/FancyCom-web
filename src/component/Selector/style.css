.screen {
  flex: 1;
  background-color: 'white';
}
.container {
  flex-direction: row;
  justify-content: center;
}
.scrollContainer {
  display: flex;
  flex-direction: row;
}
.alignItemContainer {
  padding-left: 5px;
  padding-right: 5px;
  padding-top: 10px;
  /* padding-bottom: 10px; */
}
.alignItemSection {
  justify-content: center;
  align-items: center;
}
.alignIconSection {
  width: 60px;
  height: 60px;
  border-style: solid;
  border-width: 2px;
  border-color: #DCDCDC;
  align-items: center;
  justify-content: center;
  margin-left: 5px;
  margin-right: 5px;
  display: flex;
}
.alignItemLabel {
  color: black;
  font-size: 12px;
  font-weight: bold;
  margin-top: 5px;
  text-align: center;
}
