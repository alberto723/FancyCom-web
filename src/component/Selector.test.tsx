// Canvas.test.js
import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import { Selector } from './Selector';
import BlockIcon from '@material-ui/icons/Block';
import BorderOuterIcon from '@material-ui/icons/BorderOuter';
import BorderClearIcon from '@material-ui/icons/BorderClear';

test('Should be rendered in the dom', () => {
  const selectorArray1 = [
    {
      id: 'none',
      icon: <BlockIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'NONE'
    },
    {
      id: 'border',
      icon: <BorderOuterIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'SOLID'
    },
    {
      id: 'dashed',
      icon: <BorderClearIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'DASHED'
    }
  ];

  const component = renderer.create(
    <Selector items={selectorArray1} onChange={() => console.log('onChange clicked')} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should show a list of items. The number of items should match the number of items in the array prop', () => {
  const items = [
    {
      id: 'none',
      icon: <BlockIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'NONE'
    },
    {
      id: 'border',
      icon: <BorderOuterIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'SOLID'
    },
    {
      id: 'dashed',
      icon: <BorderClearIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'DASHED'
    }
  ];

  const { queryByLabelText, getByText, container } = render(
    <Selector items={items} onChange={() => console.log('onChange clicked')} />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-alignItemContainer-3').length).toBe(items.length);
});

test('Should trigger the onChange() event when one of the items is selected', () => {
  const items = [
    {
      id: 'none',
      icon: <BlockIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'NONE'
    },
    {
      id: 'border',
      icon: <BorderOuterIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'SOLID'
    },
    {
      id: 'dashed',
      icon: <BorderClearIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'DASHED'
    }
  ];

  const { queryByLabelText, getByText, container } = render(
    <Selector items={items} onChange={() => console.log('onChange clicked')} />
  );
  container.querySelector('.makeStyles-alignItemContainer-3')?.click();

  expect(container.querySelectorAll('.makeStyles-alignItemContainer-3').length).toBe(3);
});

test('Should trigger the "onFancySelectorRender" plugin hook when the FancySelector component is rendering', () => {
  const items = [
    {
      id: 'none',
      icon: <BlockIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'NONE'
    },
    {
      id: 'border',
      icon: <BorderOuterIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'SOLID'
    },
    {
      id: 'dashed',
      icon: <BorderClearIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'DASHED'
    }
  ];

  const { queryByLabelText, getByText, container } = render(
    <Selector items={items} onChange={() => console.log('onChange clicked')} />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-alignItemContainer-3').length).toBe(items.length);
});

test('Should trigger the "onFancySelectorChange" plugin hook when an item is selected', () => {
  const items = [
    {
      id: 'none',
      icon: <BlockIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'NONE'
    },
    {
      id: 'border',
      icon: <BorderOuterIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'SOLID'
    },
    {
      id: 'dashed',
      icon: <BorderClearIcon style={{ color: '#9D9D9D', fontSize: 30 }} />,
      label: 'DASHED'
    }
  ];

  const { queryByLabelText, getByText, container } = render(
    <Selector items={items} onChange={() => console.log('onChange clicked')} />
  );
  container.querySelector('.makeStyles-alignItemContainer-3')?.click();

  expect(container.querySelectorAll('.makeStyles-alignItemContainer-3').length).toBe(items.length);
});
