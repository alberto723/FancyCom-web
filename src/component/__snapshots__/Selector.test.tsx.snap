// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be rendered in the dom 1`] = `
<div
  className="makeStyles-screen-1"
>
  <div
    className="makeStyles-container-2"
  >
    <div
      className="makeStyles-alignItemContainer-3"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#F4F4F4",
        }
      }
    >
      <div
        className="makeStyles-alignItemSection-4"
      >
        <div
          className="makeStyles-alignIconSection-5"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "#9D9D9D",
                "fontSize": 30,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM4 12c0-4.42 3.58-8 8-8 1.85 0 3.55.63 4.9 1.69L5.69 16.9C4.63 15.55 4 13.85 4 12zm8 8c-1.85 0-3.55-.63-4.9-1.69L18.31 7.1C19.37 8.45 20 10.15 20 12c0 4.42-3.58 8-8 8z"
            />
          </svg>
        </div>
        <p
          className="makeStyles-alignItemLabel-6"
        >
          NONE
        </p>
      </div>
    </div>
    <div
      className="makeStyles-alignItemContainer-3"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <div
        className="makeStyles-alignItemSection-4"
      >
        <div
          className="makeStyles-alignIconSection-5"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "#9D9D9D",
                "fontSize": 30,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v2h2V7zm0 4h-2v2h2v-2zm4 0h-2v2h2v-2zM3 3v18h18V3H3zm16 16H5V5h14v14zm-6-4h-2v2h2v-2zm-4-4H7v2h2v-2z"
            />
          </svg>
        </div>
        <p
          className="makeStyles-alignItemLabel-6"
        >
          SOLID
        </p>
      </div>
    </div>
    <div
      className="makeStyles-alignItemContainer-3"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <div
        className="makeStyles-alignItemSection-4"
      >
        <div
          className="makeStyles-alignIconSection-5"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "#9D9D9D",
                "fontSize": 30,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M7 5h2V3H7v2zm0 8h2v-2H7v2zm0 8h2v-2H7v2zm4-4h2v-2h-2v2zm0 4h2v-2h-2v2zm-8 0h2v-2H3v2zm0-4h2v-2H3v2zm0-4h2v-2H3v2zm0-4h2V7H3v2zm0-4h2V3H3v2zm8 8h2v-2h-2v2zm8 4h2v-2h-2v2zm0-4h2v-2h-2v2zm0 8h2v-2h-2v2zm0-12h2V7h-2v2zm-8 0h2V7h-2v2zm8-6v2h2V3h-2zm-8 2h2V3h-2v2zm4 16h2v-2h-2v2zm0-8h2v-2h-2v2zm0-8h2V3h-2v2z"
            />
          </svg>
        </div>
        <p
          className="makeStyles-alignItemLabel-6"
        >
          DASHED
        </p>
      </div>
    </div>
  </div>
</div>
`;
